### PR TITLE
Asb namespace naming

### PR DESCRIPTION
**Pull request by yvesgoeleven**
This PR shows how cross namespace routing can be done (now and in the past), and also explains the interactions with multinamespace partitioning

### CODE REVIEW

**Comment by yvesgoeleven** [comment]
@Particular/docs-maintainers @Particular/azure-service-bus-maintainers please review

File: Snippets/ASB/ASB_6/Usage.cs [hunk 1]
```diff
@@ -13,6 +13,17 @@ class Usage
         #endregion
     }
 
+    void NamespaceRoutingSendOptions(IBus bus)
+    {
+        string destination;
+        #region namespace_routing_send_options_full_connectionstring
+
+        destination = "sales@Endpoint=sb://destination1.servicebus.windows.net;SharedAccessKeyName=[shared access key name];SharedAccessKey=[shared access key]";
+        bus.Send(destination, new MyMessage());
```
Comment (SeanFeldman) [inline Snippets/ASB/ASB_6/Usage.cs]: Should we list the v6 way as well?
```c# 
var options = new SendOptions();
options.SetDestination("MyDestination");
await endpoint.Send(new MyMessage(), options).ConfigureAwait(false);
```

File: nservicebus/azure-service-bus/multiple-namespaces-support.md [hunk 2]
```diff
@@ -9,28 +9,33 @@ tags:
 reviewed: 2016-04-26
 ---
 
-Azure Service Bus transport enables various _namespace partitioning strategies_ to cover scenarios such as High Availability and multiple Data Center support, or to overcome [Azure services limits](https://azure.microsoft.com/en-us/documentation/articles/service-bus-quotas/).  
 
-The namespace partitioning strategy can be configured using Azure Service Bus transport configuration API.
+## Multiple namespace support
```
Comment (SeanFeldman) [inline nservicebus/azure-service-bus/multiple-namespaces-support.md]: namespace**s**

**Comment by SimonCropp** [comment]
@melkio since @SeanFeldman is on leave would you mind reviewing this?

File: Snippets/ASB/ASB_6/Usage.cs (hunk 1 above)
Comment (yvesgoeleven) [inline Snippets/ASB/ASB_6/Usage.cs]: I would limit it to 1, there are many places where you can specify an address or destination, like message endpoint mappings, unicast routing api etc... we don't want to show them all

File: nservicebus/azure-service-bus/multiple-namespaces-support.md [hunk 3]
```diff
@@ -44,3 +49,35 @@ The `FailOverNamespacePartitioning` can be used to provide High Availability. It
...
+But, this requires the namespace name and connectionstring to be registered using the `NamespaceRouting()` API.
+
+snippet: namespace_routing_registration
+
+
+### Default namespace name
+
+When using the `ConnectionString` method to configure a namespace, it will get a name as well. This name is represented by the `DefaultNamespaceName` configuration setting, which has a value of `default`.
+
+When doing cross namespace request reply communication between endpoints configured this way, in combination with the `UseNamespaceNamesInsteadOfConnectionStrings()` configuration method, then the reply address header will include a value of `"sourceendpoint@default"`. However the connectionstring that is mapped to this name is different for each endpoint in the communication and it will break the request reply pattern.
```
Comment (melkio) [inline nservicebus/azure-service-bus/multiple-namespaces-support.md]: Do we have to link the "securing connection string" somewhere?

**Comment by melkio** [comment]
With the new APIs and the possibility to configure `DefaultNamespaceName`, I think we have to review "securing connection string" article to be sure it's still valid

**Comment by indualagarsamy** [comment]
@yvesgoeleven @melkio - are more changes to this PR?

**Comment by indualagarsamy** [comment]
@yvesgoeleven @melkio -  ping. Can you please let us know what's remaining on this PR? I'd like for this PR not to get too stale, requiring a rebase, etc.

**Comment by melkio** [comment]
Sorry, put some comments. Waiting for @yvesgoeleven reply

**Comment by melkio** [comment]
It's fine by me. Ready to be merged :+1:

File: Snippets/ASB/ASB_6/Usage.cs (hunk 1 above)
Comment (SeanFeldman) [inline Snippets/ASB/ASB_6/Usage.cs]: In that case, would be nice if we could have a sample that use one way, but the text linked to the all possible ways (if that doco exists).

Comment (yvesgoeleven) [inline Snippets/ASB/ASB_6/Usage.cs]: links added

File: nservicebus/azure-service-bus/multiple-namespaces-support.md (hunk 3 above)
Comment (yvesgoeleven) [inline nservicebus/azure-service-bus/multiple-namespaces-support.md]: link added

**Pull request merged**